tests/graph/test_op.py

```diff
@@ -232,3 +232,55 @@ def perform(self, *_):
 
     x = pt.TensorType(dtype="float64", shape=(1,))("x")
     assert SomeOp()(x).type == pt.dvector
+
+
+def test_op_name():
+    op_name = "op_name"
+
+    class DummyType(Type):
+        def filter(self, data):
+            return data
+
+        def __eq__(self, other):
+            return isinstance(other, DummyType)
+
+        def __hash__(self):
+            return hash(DummyType)
+
+        def __repr__(self):
+            return "DummyType()"
+
+    def dummy_variable(name):
+        return Variable(DummyType(), None, None, name=name)
+
+    x = dummy_variable("x")
+
+    class SingleOutOp(Op):
+        def make_node(self, *inputs):
+            outputs = [dummy_variable("a")]
+            return Apply(self, list(inputs), outputs)
+
+        def perform(self, node, inputs, outputs):
+            raise NotImplementedError()
+
+    single_op = SingleOutOp()
+    res_single = single_op(x, name=op_name)
+    assert res_single.name == op_name
+
+    class MultiOutOp(Op):
+        def make_node(self, *inputs):
+            outputs = [dummy_variable("a"), dummy_variable("b")]
+            return Apply(self, list(inputs), outputs)
+
+        def perform(self, node, inputs, outputs):
+            raise NotImplementedError()
+
+    multi_op = MultiOutOp()
+    res = multi_op(x, name=op_name)
+    for i, r in enumerate(res):
+        assert r.name == f"{op_name}_{i}"
+
+    multi_op = MultiOutOp()
+    multi_op.default_output = 1
```

Comment:
A bit of a nitpick, but more "realistic" if you make the default_output something defined when a class is initialized. Just need to define a `__init__` method that assigns it to self after calling super().__init__.

```suggestion
    multi_op = MultiOutOp(default_output=1)
```

Reply:
That way you can also parametrize the number of outputs, with a `multi_output=False|True` that reduces the lines of code in this test just a tiny bit. Then in `made_node`, you can do:

```python
if self.multi_output:
  outputs = [input.type(), input.type()]
else:
  outputs = [input.type()]
```